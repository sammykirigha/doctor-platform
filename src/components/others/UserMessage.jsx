import React from "react";
import { GoPrimitiveDot } from "react-icons/go";
import { MdOutlineAvTimer } from "react-icons/md";
import image1 from "../../data/images/01.jpg"

const UserMessage = ({index}) => {
    return (
        <div className={`flex flex-row ml-3 ${index % 2 === 0 && 'justify-start flex-row-reverse mr-2 gap-2'}`}>
            <div className=" mt-6 relative h-12 w-12 bg-blue-100 flex items-center justify-center rounded-full ">
                <img
                    src={image1}
                    alt="logo"
                    className="m-2 w-12 h-12 rounded-full cursor-pointer "
                />
                
                <span className="absolute -top-2 right-7">
                    <GoPrimitiveDot className="text-green-700 h-6 w-6 " />{" "}
                </span>
            </div>
            <div className="max-w-[60%] bg-white flex flex-col mt-5 ml-2 ">
                <h6 className="text-gray-400 border border-gray-300 mt-3 h-[35px] pt-1 pl-1 rounded-md px-2">
                    Nice to meet you
                </h6>
                <div className="flex flex-row items-center">
                    <MdOutlineAvTimer className="h-4 w-4 text-gray-500" />
                    <span className="text-gray-500">42 min ago</span>
                </div>
            </div>
        </div>
    );
};

export default UserMessage;
