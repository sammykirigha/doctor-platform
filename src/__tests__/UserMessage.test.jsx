/* eslint-disable testing-library/prefer-screen-queries */
import { cleanup, render, screen } from '@testing-library/react';
import UserMessage from '../components/UserMessage'


describe('The suit is to test the body component', () => {

	it('Snapshot of UserMessage', () => {
		const { asFragment } = render(<UserMessage index={'1'} />)
		expect(asFragment()).toMatchSnapshot()
	})

	it('should render the UserMessage with the right index', () => {
		render(<UserMessage />)
		let element = screen.getByText('Nice to meet you');

		expect(element.textContent).toBe('Nice to you')

	})

})