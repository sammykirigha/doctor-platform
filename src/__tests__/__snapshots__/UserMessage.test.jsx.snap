// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The suit is to test the body component Snapshot of UserMessage 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row ml-3 false"
  >
    <div
      class=" mt-6 relative h-12 w-12 bg-blue-100 flex items-center justify-center rounded-full "
    >
      <img
        alt="logo"
        class="m-2 w-12 h-12 rounded-full cursor-pointer "
        src="01.jpg"
      />
      <span
        class="absolute -top-2 right-7"
      >
        <svg
          class="text-green-700 h-6 w-6 "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 8 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z"
            fill-rule="evenodd"
          />
        </svg>
         
      </span>
    </div>
    <div
      class="max-w-[60%] bg-white flex flex-col mt-5 ml-2 "
    >
      <h6
        class="text-gray-400 border border-gray-300 mt-3 h-[35px] pt-1 pl-1 rounded-md px-2"
      >
        Nice to meet you
      </h6>
      <div
        class="flex flex-row items-center"
      >
        <svg
          class="h-4 w-4 text-gray-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 17c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1zm0-14v4h2V5.08c3.39.49 6 3.39 6 6.92 0 3.87-3.13 7-7 7s-7-3.13-7-7c0-1.68.59-3.22 1.58-4.42L12 13l1.41-1.41-6.8-6.8v.02C4.42 6.45 3 9.05 3 12c0 4.97 4.02 9 9 9a9 9 0 000-18h-1zm7 9c0-.55-.45-1-1-1s-1 .45-1 1 .45 1 1 1 1-.45 1-1zM6 12c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1z"
          />
        </svg>
        <span
          class="text-gray-500"
        >
          42 min ago
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
